node.go

```diff
@@ -135,53 +53,65 @@ func (n *SphinxNode) PrefixFreeDecode(s []byte) (int, []byte, []byte) {
 	return Failure, nil, nil
 }
 
-// Unwrap unwraps a layer of encryption from a sphinx packet
-// and upon success returns an UnwrappedMessage, otherwise an error.
-func (n *SphinxNode) Unwrap(packet *SphinxPacket) (*UnwrappedMessage, error) {
+// UnwrappedMessage is produced by SphinxNode's Unwrap method
+type UnwrappedMessage struct {
+	ProcessAction int
+	Alpha         []byte
+	Beta          []byte
+	Gamma         []byte
+	Delta         []byte
+	NextHop       []byte
+	ClientID      []byte
+	MessageID     []byte
+}
+
+// SphinxNodeState is an interface for sphinx mix node state
+type SphinxNodeState interface {
```

Comment:
yes that's correct. should we instead have a replace_cache object and a private_key object?